epochX/cudacpp/CODEGEN/PLUGIN/CUDACPP_SA_OUTPUT/launch_plugin.py

```diff
@@ -85,14 +93,23 @@ def plugin_input(self, finput):
 
     def default_setup(self):
         super().default_setup()
-        self.add_param('floating_type', 'm', include=False, hidden=True,
-                       fct_mod=(self.reset_makeopts,(),{}),
-                       allowed=['m','d','f'],
-                       comment='floating point precision: f (single), d (double), m (mixed: double for amplitudes, single for colors)'
-                       )
         cudacpp_supported_backends = [ 'fortran', 'cuda', 'hip', 'cpp', 'cppnone', 'cppsse4', 'cppavx2', 'cpp512y', 'cpp512z', 'cppauto' ]
-        self.add_param('cudacpp_backend', 'cpp', include=False, hidden=False,
+        self.add_param('cudacpp_backend', 'cpp',
+                       include=False, # AV: 'include=True' would add "CUDACPP_BACKEND = 'cpp'" to run_card.inc
+                       hidden=False, # AV: keep cudacpp_backend in runcard template and keep 'hidden='False'
                        allowed=cudacpp_supported_backends)
+        self.add_param('cudacpp_fptype', 'm',
+                       include=False, # AV: 'include=True' would add "CUDACPP_FPTYPE = 'm'" to run_card.inc (if fct_mod is removed, else codegen fails)
+                       hidden=False, # AV: add cudacpp_backend to runcard template and keep 'hidden='False'
+                       fct_mod=(self.reset_makeopts,(),{}), # AV: I assume this forces a 'make cleanavx' if FPTYPE changes?
+                       allowed=['m','d','f']
+                       )
+        self.add_param('cudacpp_bldall', 'False',
```

Comment:
A boolean should be 
self.add_param('cudacpp_bldall', False,
and does not need an allowed (since F T .false. .true. false False true True 0 1 are all valid entry for boolean.

Reply:
Hi @oliviermattelaer thanks. As mentioned above, I make these changes in PR #948 which is smaller and more manageable, so hopefully can be reviewed/approved more easily. I changed to boolean in https://github.com/madgraph5/madgraph4gpu/pull/948/commits/c68a9a7dcd282b1f7114e5c5db5cf937fd47228f
Marking this as resolved